### PR TITLE
Search SDK: Test code changes

### PR DESCRIPTION
**Pull request by bitsforthought**
See individual commits for details.

FYI @chaosrealm @Yahnoosh

### CODE REVIEW

File: src/Search/Search.Tests/Tests/IndexManagementTests.cs [hunk 1]
```diff
@@ -205,9 +205,7 @@ public void DeleteIndexIfNotChangedWorksOnlyOnCurrentResource()
             });
         }
 
-        // TODO,seansaleh: Enable this test once the server side bug is fixed
-        [Fact(Skip = "There is a bug in Azure Search that 404's before checking the ETag. "
-           + "This test exposes this bug and thus is disabled until the server side issue is fixed")]
+        [Fact]
         public void DeleteIndexIfExistsWorksOnlyWhenResourceExists()
```
Comment (chaosrealm) [inline src/Search/Search.Tests/Tests/IndexManagementTests.cs]: "Works" is meaningless. Can test more precisely describe what is the expected behavior? I.e. Throws 404 or Returns Success

File: src/Search/Search.Tests/Tests/IndexerTests.cs [hunk 2]
```diff
@@ -190,9 +190,7 @@ public void DeleteIndexerIfNotChangedWorksOnlyOnCurrentResource()
             });
         }
 
-        // TODO,seansaleh: Enable this test once the server side bug is fixed
-        [Fact(Skip = "There is a bug in Azure Search that 404's before checking the ETag. "
-           + "This test exposes this bug and thus is disabled until the server side issue is fixed")]
+        [Fact]
         public void DeleteIndexerIfExistsWorksOnlyWhenResourceExists()
```
Comment (chaosrealm) [inline src/Search/Search.Tests/Tests/IndexerTests.cs]: Ditto

**Comment by chaosrealm** [comment]
LGTM modulo naming of those two tests

**Comment by brjohnstmsft** [comment]
@chaosrealm I'll rename them in a subsequent PR

**Pull request merged**